frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +75,56 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and kind
+ */
+export const useModelFinder = () => {
+  const models: K8sKind[] = _.values(
+    useSelector(({ k8s }) => k8s.getIn(['RESOURCES', 'models'])).toJS(),
+  );
+  const groupVersionMap: DiscoveryResources['groupVersionMap'] = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'groupToVersionMap']),
+  );
+
+  const findModel = (group: string, kind: string) => {
+    const groupObject = _.find(groupVersionMap, (item) => item.name === group);
+    const { preferredVersion, versions } = groupObject || {};
+    let model = undefined;
```

Comment:
```suggestion
    let model: K8sKind;
```

Reply:
Avoid having `model`, `model2` vars outside of each `if`. It should be necessary. You should be able to do something like

```ts
    if (preferredVersion) {
      // Find a model for the CRD that uses this preferred version
      const model = _.find(...);
      if (model) {
        return model;
      }
    }
```